app/models/user.rb

```diff
@@ -25,6 +25,13 @@ def email_activate
     update(active: true, email_token: nil)
   end
 
+  def bookmarks
+    videos.order(:position).each_with_object({}) do |video, acc|
+      acc[video.tutorial_id] = [] if acc[video.tutorial_id].nil?
+      acc[video.tutorial_id] << video
+    end
+  end
```

Comment:
I bet I can improve this (e.g., getting the tutorial title in the same query), so I'm going to keep working on this.